### ISSUE TITLE
UUID foreign keys no longer work in Laravel 6

### ISSUE THREAD

**Issue by edeis53** [michaeldyrynda/laravel-model-uuid#65]
**TLDR reproduction at the bottom**

We've previously been using `laravel-model-uuid` to have all our IDs as binary uuid. Which worked quite well and FK relations worked too.

Packages we were using:
```
        "dyrynda/laravel-efficient-uuid": "^2.2",
        "dyrynda/laravel-model-uuid": "~4.1",
        "laravel/framework": "5.7.*",
```
With this configuration, all we had to do is set the following casts on our models and everything worked brilliantly. 
```
    public function uuidColumn()
    {
        return 'id';
    }
    protected $casts = [
        'uuid' => 'uuid', 
    ];
```

But since upgrading the packages and Laravel, we can save the related records using binary uuid, but after saving, if you try to find or get the relationship, it comes up empty, even though you can see it in the database with the correct foreign keys.

Packages:
```
        "dyrynda/laravel-efficient-uuid": "^3.1",
        "dyrynda/laravel-model-uuid": "^5.0",
        "laravel/framework": "^6.4",
```

Using the current packages, we have to cast the foreign key too, otherwise we get an error
`SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'user_id'`

and have adjusted the code on the models to the following:
```
    protected $casts = ['id' => 'uuid', 'user_id' => 'uuid'];

    public $incrementing = false;

    public function uuidColumns(): array
    {
        return ['id', 'user_id'];
    }
```

We can create records and create related records via foreign keys just fine. They all show up in the database as expected. However, after the records have been created, laravel can't find any related models, eg. `user.hasMany(posts)` 

```
Created # of users: 1
Created # of posts: 3
Found # of user->posts relations: 0
```
I think maybe it's comparing the cast to string UUID against the binary column in the database and can't find a match.

TLDR;
I've made a detailed reproduction here.
https://github.com/edeis53/laravel6-uuid-issue

All you need to do is clone the repo, `composer install`, and then `php artisan migrate:fresh --seed`. There's a small test in the seed that outputs to console.

**Comment by edeis53** [comment]
Actually, the cast/binary mismatch is exactly what is happening.

You can see the `post` record being created with a binary `id` and foreign key `user_id`
```
[2019-11-27 05:39:57] local.INFO: insert into `posts` (`title`, `user_id`, `id`, `updated_at`, `created_at`) values (?, ?, ?, ?, ?)  
[2019-11-27 05:39:57] local.INFO: array (
  0 => 'Veniam error quod minima. Error laborum voluptatum sunt exercitationem dolor. Quis qui ipsa occaecati iste minus dolor omnis.',
  1 => '=��K�WE3��wQ
��',
  2 => '�7n��kLS�T$�L�',
  3 => '2019-11-27 05:39:57',
  4 => '2019-11-27 05:39:57',
```

But if you try to query the relationship, eg. `$user->posts` the cast value of `user_id` is used.
```
[2019-11-27 05:39:57] local.INFO: select * from `posts` where `posts`.`user_id` = ? and `posts`.`user_id` is not null  
[2019-11-27 05:39:57] local.INFO: array (
  0 => '3d93a84b-fb57-4533-96ab-1c77510ab4e0',
)  
```

**Comment by edeis53** [comment]
Going to close this. I think in our 5.7 branch we we're actually casting the binary uuid and they were staying as binary, hence eloquent having no problem with foreign relations.
